packages/babel-plugin-transform-for-of/src/index.ts

```diff
@@ -20,9 +20,9 @@ function buildLoopBody(
   const body = newBody ?? bodyPath.node;
   if (
     t.isBlockStatement(body) &&
-    Object.keys(path.getBindingIdentifiers()).some(id =>
-      bodyPath.scope.hasOwnBinding(id),
-    )
+    Object.keys(path.getBindingIdentifiers())
+      .concat(Object.keys(path.get("right").getBindingIdentifiers()))
```

Comment:
Should we fix this in `getBindingIdentifiers`?

Reply:
I'm not 100% sure about what's happening here. `right` does not introduce new bindings, because it cannot contain variable declarations.

Reply:
I think the fix should be to force memoization in https://github.com/babel/babel/blob/a345a102f6cc1d384946b165d6375f503f2ae2a3/packages/babel-plugin-transform-for-of/src/index.ts#L93 if `t.isIdentifier(right)` and `body.scope.hasOwnBinding(right.name)`.